api/src/test/java/org/openmrs/api/VisitServiceTest.java

```diff
@@ -938,21 +937,25 @@ public void saveVisit_shouldAssociateEncounterWithVisitOnSaveEncounter() {
 	/**
 	 * @see VisitService#saveVisit(Visit)
 	 */
```

Comment:
Hibernate throws this exception for the other side of the relationship i.e. if you save an encounter where visit is transient and not the opposite